protos/core.proto

```diff
@@ -11,6 +11,8 @@ message ProjectConfig {
   string default_schema = 2;
   string assertion_schema = 5;
 
+  string schema_suffix = 7;
```

Comment:
do we expect users to use this in dataform.json? i'm worried it's a bit confusing. (why would the user set this, given the `default_schema` and `assertion_schema` settings?)

perhaps the answer is that we imagine allowing the user to have multiple dataform.json files at different levels in the directory hierarchy one day?

Reply:
It was more that as we add more settings and overrides, everything is stored in a single object. I don't need to now separately pass a a compileOptions object all the way into the core runtime through genIndex etc.

Reply:
yep understood, just would rather not have this here if we can help it (for API usability)...

FWIW if this were just passed in to compile and not part of ProjectConfig, we could just add the suffixes as part of the compile process, and wouldn't need to touch dataform/core.

Reply:
Let's discuss, as I think we are going to very soon add a number of new CLI options that will override projectConfig settings (even the warehouse property which would be very useful for testing) and would like a way to do that without not too much hassle.